templates/SilverStripe/Admin/Includes/LeftAndMain_ViewModeSelector.ss

```diff
@@ -2,7 +2,7 @@
     <%-- Placeholder is re-rendered by the ShareDraftContent React component --%>
     <span class="share-draft-content__placeholder"
         data-url="<% if $CurrentPage.ShareDraftLinkAction %>{$CurrentPage.ShareDraftLinkAction}<% else %>{$Controller.CurrentPage.ShareDraftLinkAction}<% end_if %>"
```

Comment:
Note these references must _not_ be updated to `currentRecord`.

`CurrentPage` is not actually accessing the `CurrentPage()` method on the `LeftAndMain` (or in this case `CMSMainPageEditController`) instance. Instead, because of how this template gets rendered, it's falling back to the global template provider `$CurrentPage` which calls `Controller::curr()`.

`ShareDraftLinkAction()` then gets called _on the controller_, not on the page record.